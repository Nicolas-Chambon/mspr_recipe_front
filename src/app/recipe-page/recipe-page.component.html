<div class="container" *ngIf="recipes$ | async as recipesList">
  <h2 mat-subheader>Choisir des recettes :</h2>
  <mat-divider></mat-divider>
  <mat-selection-list multiple [(ngModel)]="selectedOptions">
    <mat-list-option *ngFor="let recipe of recipesList" [value]="recipe.id">
      {{ recipe.name }}
    </mat-list-option>
  </mat-selection-list>
  <mat-divider></mat-divider>
  <div class="button-container">
    <button mat-stroked-button color="primary" [disabled]="selectedOptions.length <= 0" (click)="downloadRecipes()">
      Télécharger les recettes
    </button>
  </div>
</div>
