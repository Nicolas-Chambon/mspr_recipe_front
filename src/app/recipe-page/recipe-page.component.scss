.container {
  margin: 10px 100px;
}

.button-container {
  display: flex;
  justify-content: flex-end;
  margin-top: 30px;
}
