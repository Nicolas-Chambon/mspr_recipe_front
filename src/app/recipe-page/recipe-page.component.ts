import { Component, OnInit } from '@angular/core';
import { RecipeApiService } from '../core/services/recipe-api.service';
import { Observable } from 'rxjs';

@Component({
  selector: 'app-recipe-page',
  templateUrl: './recipe-page.component.html',
  styleUrls: ['./recipe-page.component.scss'],
})
export class RecipePageComponent implements OnInit {
  recipes$: Observable<any>;

  selectedOptions = [];

  constructor(private recipeService: RecipeApiService) {}

  ngOnInit(): void {
    this.recipes$ = this.recipeService.list();
  }
}
