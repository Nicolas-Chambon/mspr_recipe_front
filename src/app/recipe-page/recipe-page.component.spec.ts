import { ComponentFixture, TestBed } from '@angular/core/testing';

import { RecipePageComponent } from './recipe-page.component';
import { RouterTestingModule } from "@angular/router/testing";
import { MaterialModule } from "../shared/material.module";
import { FormsModule } from "@angular/forms";
import {RecipeApiService} from "../core/services/recipe-api.service";
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { HttpClientTestingModule } from '@angular/common/http/testing';

describe('RecipePageComponent', () => {
  let component: RecipePageComponent;
  let fixture: ComponentFixture<RecipePageComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      providers: [RecipeApiService],
      imports: [
        RouterTestingModule.withRoutes([]),
        MaterialModule,
        FormsModule,
        HttpClientTestingModule,
        BrowserAnimationsModule
      ],
      declarations: [RecipePageComponent],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(RecipePageComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
