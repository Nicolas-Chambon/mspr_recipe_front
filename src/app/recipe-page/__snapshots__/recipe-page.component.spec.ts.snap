// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipePageComponent should equal snapshot 1`] = `
<app-recipe-page
  recipeService={[Function RecipeApiService]}
  recipes$={[Function Observable]}
  selectedOptions={[Function Array]}
>
  
</app-recipe-page>
`;
